### PR TITLE
Stark continuous read-only memory example

### PR DESCRIPTION
**Pull request by ColoCarletti**
# Add Stark example for continuous read-only memory constraints

- [x] New feature
- [ ] Bug fix
- [ ] Optimization

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run

### CODE REVIEW

File: provers/stark/src/examples/read_only_memory.rs [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    ) {
+        let first_step = frame.get_evaluation_step(0);
+        let second_step = frame.get_evaluation_step(1);
+
+        let a_sorted0 = first_step.get_main_evaluation_element(0, 2);
+        let a_sorted1 = second_step.get_main_evaluation_element(0, 2);
+        // (a'_{i+1} - a'_i)(a'_{i+1} - a'_i - 1) = 0 where a' is the sorted address
+        let res = (a_sorted1 - a_sorted0) * (a_sorted1 - a_sorted0 - FieldElement::<F>::one());
+
+        transition_evaluations[self.constraint_idx()] = res;
```
Comment (pablodeymo) [inline provers/stark/src/examples/read_only_memory.rs:68]: you should check if the vector can hold the value at that position

File: provers/stark/src/examples/read_only_memory.rs [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        let a_sorted = &main_segment_cols[2];
+        let v_sorted = &main_segment_cols[3];
+        let z = &challenges[0];
+        let alpha = &challenges[1];
+
+        let trace_len = trace.num_rows();
+
+        let mut aux_col = Vec::new();
+        let num = z - (&a[0] + alpha * &v[0]);
+        let den = z - (&a_sorted[0] + alpha * &v_sorted[0]);
```
Comment (diegokingston) [inline provers/stark/src/examples/read_only_memory.rs:268]: For learning purposes this is ok, but for performance reasons, it's best to compute all the denominators, due batch inverse and then compute

**Review by diegokingston — APPROVED** [review]

File: provers/stark/src/examples/read_only_memory.rs [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+        let res = (a_sorted_1 - a_sorted_0) * (a_sorted_1 - a_sorted_0 - FieldElement::<F>::one());
+
+        // The eval always exists, except if the constraint idx were incorrectly defined.
+        if let Some(eval) = transition_evaluations.get_mut(self.constraint_idx()) {
+            *eval = res;
+        }
+    }
+}
+
+#[derive(Clone)]
```
Comment (pablodeymo) [inline provers/stark/src/examples/read_only_memory.rs:75]: this annotation should be placed after the documentation

Comment (nicole-graus) [reply provers/stark/src/examples/read_only_memory.rs]: ok, [done](https://github.com/lambdaclass/lambdaworks/pull/940/commits/59c8b2681bf85c79d3bd5c484569d4ceb3a63612).

**Review by pablodeymo — APPROVED** [review]

**Pull request merged**